sketch_map_tool/tasks.py

```diff
@@ -92,24 +94,35 @@ def georeference_sketch_maps(
     uuids: list[str],
     map_frames: dict[str, NDArray],
     bboxes: list[Bbox],
+    layers: list[Layer],
 ) -> AsyncResult | BytesIO:
-    def process(sketch_map_id: int, uuid: str, bbox: Bbox) -> BytesIO:
-        """Process a Sketch Map."""
+    def process(
+        sketch_map_id: int,
+        uuid: str,
+        bbox: Bbox,
+        attribution: str,
+    ) -> list[BytesIO]:
+        """Process a Sketch Map and its attribution."""
         # r = interim result
         r = db_client_celery.select_file(sketch_map_id)
         r = to_array(r)
         r = clip(r, map_frames[uuid])
         r = georeference(r, bbox)
-        return r
+        attribution = re.sub("<.*?>", "\n", attribution)
+        attribution_info = BytesIO(attribution.encode())
+        return [r, attribution_info]
 
-    def zip_(file: BytesIO, file_name: str):
+    def zip_(file: list[BytesIO], file_name: str):
         with ZipFile(buffer, "a") as zip_file:
             name = ".".join(file_name.split(".")[:-1])
-            zip_file.writestr(f"{name}.geotiff", file.read())
+            zip_file.writestr(f"{name}.geotiff", file[0].read())
+            zip_file.writestr("attributions.txt", file[1].read())
 
     buffer = BytesIO()
-    for file_id, uuid, bbox, file_name in zip(file_ids, uuids, bboxes, file_names):
-        zip_(process(file_id, uuid, bbox), file_name)
+    for file_id, uuid, bbox, layer, file_name in zip(
+        file_ids, uuids, bboxes, layers, file_names
+    ):
+        zip_(process(file_id, uuid, bbox, get_attribution(layer)), file_name)
```

Comment:
I would add the attribution.txt after the for loop to the buffer.

Reply:
done